eos/utils/interpolation.hh

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2022 Philip Lüghausen
+ *
+ * This file is part of the EOS project. EOS is free software;
+ * you can redistribute it and/or modify it under the terms of the GNU General
+ * Public License version 2, as published by the Free Software Foundation.
+ *
+ * EOS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#ifndef EOS_GUARD_EOS_UTILS_INTERPOLATION_HH
+#define EOS_GUARD_EOS_UTILS_INTERPOLATION_HH 1
+
+#include <vector>
+#include <functional>
+#include <memory>
+#include <gsl/gsl_spline.h>
+
+namespace eos
+{
+    class CSplineInterpolation
+    {
+        public:
+            CSplineInterpolation() = delete;
+            CSplineInterpolation(const std::vector<double> & data_x, const std::vector<double> & data_y,
+                std::function<double (const double &)> map_x = [](const double & x){return x;});
+
+            /*!
+             * Evaluate with operator()
+             * Instance can be called like a function, i.e. is a functor
```

Comment:
That is not a helpful comment. That is the very meaning of having an ``operator()``. I suggest:
```c++
/*!
 * Evaluate the interpolating function.
 *
 * @param x The point at which the function shall be evaluated.
 */
```